### ISSUE TITLE
Incorrect calculation of running loss in PyTorch example code

### ISSUE THREAD

**Issue by kuihao** [adap/flower#1604]
### Describe the bug

Dear maintainers,

I am a loyal user of **Flwr + Tensorflow**, and recently started using **Flwr + PyTorch**. However, I have noticed that the [Flwr example: quickstart_pytorch](https://github.com/adap/flower/blob/main/examples/quickstart_pytorch/client.py#L54) does not calculate the loss correctly.

According to the official [PyTorch CrossEntropyLoss documentation](https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html#torch.nn.CrossEntropyLoss), `torch.nn.CrossEntropyLoss` is calculated by default as `reduction='mean'`, meaning the function outputs the "mean of the minibatch" obtained from each iteration of the dataloader. So to calculate `loss per epoch` directly, it should be `running_loss / len(dataloader)`, because `len(dataloader)` gives "how many steps of dataloader (number of minibatches)".

The current algorithm in the [example code: quickstart_pytorch](https://github.com/adap/flower/blob/main/examples/quickstart_pytorch/client.py#L54):
```python 
def test(net, testloader):
    """Validate the model on the test set."""
    criterion = torch.nn.CrossEntropyLoss()
    correct, total, loss = 0, 0, 0.0
    with torch.no_grad():
        for images, labels in tqdm(testloader):
            outputs = net(images.to(DEVICE))
            labels = labels.to(DEVICE)
            loss += criterion(outputs, labels).item()
            total += labels.size(0)
            correct += (torch.max(outputs.data, 1)[1] == labels).sum().item()
    return loss / len(testloader.dataset), correct / total
```
To correct this, `running_loss / len(dataloader.dataset)` should be applied to `torch.nn.CrossEntropyLoss(reduction='sum')` to calculate the average of the entire dataset losses.

Addendum: 
- The term "**running_loss**" is a reference from the [PyTorch official tutorial](https://pytorch.org/tutorials/beginner/introyt/tensorboardyt_tutorial.html#graphing-scalars-to-visualize-training), which refers to the **accumulation of loss values during training**.
-  It should be noted that there are different algorithms for calculating running loss, such as in the code for the [PyTorch official tutorial page (Tutorials > PyTorch TensorBoard Support)](https://pytorch.org/tutorials/beginner/introyt/tensorboardyt_tutorial.html#graphing-scalars-to-visualize-training), which **calculates the average running loss** once for a **fixed** number of update steps. The same method can be found on the [tutorial page of nvflare](https://github.com/NVIDIA/NVFlare/blob/b4ce3a8faec19363574fb619329668b5cf926490/examples/hello-pt/app/custom/cifar10trainer.py#L149), a tool from a different organization.

It should be noted that while this issue with the calculation of running loss may not affect the computation of gradients using autograd, it could have an impact on literature research and experimental data comparison. If this is determined to be a problem, I am willing to provide a fix for the PyTorch example code (e.g. 
[flower: quickstart_pytorch/client.py](https://github.com/adap/flower/blob/main/examples/quickstart_pytorch/client.py#L54) & [flower: simulation_pytorch/utils.py](https://github.com/adap/flower/blob/main/examples/simulation_pytorch/utils.py)). 

Sincerely, Kuihao.

### Steps/Code to Reproduce

The code of [quickstart_pytorch](https://github.com/adap/flower/blob/main/examples/quickstart_pytorch/client.py#L54)
```python
def test(net, testloader):
    """Validate the model on the test set."""
    criterion = torch.nn.CrossEntropyLoss()
    correct, total, loss = 0, 0, 0.0
    with torch.no_grad():
        for images, labels in tqdm(testloader):
            outputs = net(images.to(DEVICE))
            labels = labels.to(DEVICE)
            loss += criterion(outputs, labels).item()
            total += labels.size(0)
            correct += (torch.max(outputs.data, 1)[1] == labels).sum().item()
    return loss / len(testloader.dataset), correct / total
```

### Expected Results

I guess what Flwr was trying to get was "the average of the losses values of the whole data set".

The code should probably be fixed as follows:
```python
def test(net, testloader):
    """Validate the model on the test set."""
    criterion = torch.nn.CrossEntropyLoss(reduction='sum')
    correct, loss = 0, 0.0
    with torch.no_grad():
        for images, labels in tqdm(testloader):
            outputs = net(images.to(DEVICE))
            labels = labels.to(DEVICE)
            loss += criterion(outputs, labels).detach()
            correct += (torch.max(outputs.data, 1)[1] == labels).sum().detach()
    return loss / len(testloader.dataset), correct / len(testloader.dataset)
```
- `loss / len(dataloader.dataset)` should be applied to `torch.nn.CrossEntropyLoss(reduction='sum')`
- `total` doesn't need to be added up in each iteration, just divide `correct` by `len(testloader.dataset)` at the final return to improve performance

### Actual Results

The code actually gets a **much smaller** loss value than expected because the current algorithm sums multiple "minibatch averages" and then divides by the "size of the entire data set".
